esp-wifi/src/common_adapter/common_adapter_esp32c2.rs

```diff
@@ -30,27 +33,9 @@ pub(crate) unsafe fn phy_enable() {
                 super::phy_enable_clock();
             }
 
-            if !unsafe { G_IS_PHY_CALIBRATED } {
-                let mut cal_data: [u8; core::mem::size_of::<esp_phy_calibration_data_t>()] =
-                    [0u8; core::mem::size_of::<esp_phy_calibration_data_t>()];
-
-                let phy_version = unsafe { get_phy_version_str() };
-                trace!("phy_version {}", unsafe { str_from_c(phy_version) });
-
-                let init_data = &PHY_INIT_DATA_DEFAULT;
-
-                // DON'T CALL `phy_bbpll_en_usb` on ESP32-C2
-
-                unsafe {
-                    register_chipv7_phy(
-                        init_data,
-                        &mut cal_data as *mut _
-                            as *mut crate::binary::include::esp_phy_calibration_data_t,
-                        esp_phy_calibration_mode_t_PHY_RF_CAL_FULL,
-                    );
-
-                    G_IS_PHY_CALIBRATED = true;
-                }
+            if unsafe { !G_IS_PHY_CALIBRATED } {
+                super::phy_calibrate();
+                unsafe { G_IS_PHY_CALIBRATED = true };
```

Comment:
Do we not need any kind of synchronization here?

Reply:
should use atomic 👍

Reply:
We're inside a critical section, you can use the cheaper `critical_section::Mutex`

Reply:
Ah, GH decided to hide the two CS lines. I guess this doesn't need anything more? Unless `G_IS_PHY_CALIBRATED` is readable outside of init?

Reply:
Let's start getting rid of mutable statics :)

Reply:
We can clean this up later, it seems sound at least from where I'm looking.